connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SourceTaskOffsetCommitter.java

```diff
@@ -67,70 +69,53 @@ public void close(long timeoutMs) {
     public void schedule(final ConnectorTaskId id, final WorkerSourceTask workerTask) {
         synchronized (committers) {
             long commitIntervalMs = config.getLong(WorkerConfig.OFFSET_COMMIT_INTERVAL_MS_CONFIG);
-            ScheduledFuture<?> commitFuture = commitExecutorService.schedule(new Runnable() {
+            ScheduledFuture<?> commitFuture = commitExecutorService.scheduleWithFixedDelay(new Runnable() {
                 @Override
                 public void run() {
                     commit(id, workerTask);
                 }
-            }, commitIntervalMs, TimeUnit.MILLISECONDS);
-            committers.put(id, new ScheduledCommitTask(commitFuture));
+            }, 0, commitIntervalMs, TimeUnit.MILLISECONDS);
+            committers.put(id, commitFuture);
```

Comment:
I'll update the code to depict my suggestion. 

The suggestion is that commit shouldn't check whether the task has been cancelled by executing a `get` on the map. If `commit` succeeds means the task has not been cancelled yet. Any races between threads that try to remove tasks and threads that try to start those tasks, should be resolved with implicit barriers. You can't successfully remove a task that hasn't started yet only with mutual exclusion (and this refers to both this and the previous implementations).